### PR TITLE
Add Other Metagames of the Month July '18

### PR DESCRIPTION
**Pull request by Spandamn**
Along with a command for reversed
@TheImmortal

### CODE REVIEW

File: config/formats.js [hunk 1]
```diff
@@ -420,65 +420,62 @@ let Formats = [
...
-					if (pokemon.prevo) {
-						pokemon = this.getTemplate(pokemon.prevo);
-					}
-				}
-				if (!families[pokemon.species]) families[pokemon.species] = [];
-				families[pokemon.species].push(set.species);
+		mod: 'gen7',
+		ruleset: ['[Gen 7] OU'],
+		banlist: ['Dragonite', 'Kyurem-Black'],
+		onModifyMovePriority: -1,
```
Comment (urkerab) [inline config/formats.js]: If you change this to `-2` then you don't have to worry about the `-ate`s, although you still have to check for `Normalize` I guess.

File: config/formats.js [hunk 2]
```diff
@@ -420,65 +420,62 @@ let Formats = [
...
+		onModifyMove: function (move, pokemon) {
+			if (move.type === 'Normal' && !move.isZ && move.id !== 'hiddenpower' && !pokemon.hasAbility(['aerilate', 'galvanize', 'normalize', 'pixilate', 'refrigerate'])) {
```
Comment (urkerab) [inline config/formats.js]: I think Hidden Power's type has already changed by this point, although Natural Gift's type might not have, not that anyone uses it.

File: config/formats.js [hunk 3]
```diff
@@ -420,65 +420,62 @@ let Formats = [
...
 			}
-			return problems;
+			return this.validateSet(set, teamHas, forgedTemplate);
 		},
 	},
 	{
-		name: "[Gen 7] Scalemons",
-		desc: `Every Pok&eacute;mon's stats, barring HP, are scaled to give them a BST as close to 600 as possible.`,
+		name: "[Gen 7] Reversed",
+		desc: `Every Pok&eacute;mon has ALL of it's base Physical and Special stats swapped.`,
```
Comment (urkerab) [inline config/formats.js]: its

**Comment by Spandamn** [comment]
Any clue on how to fix the build?

**Comment by Zarel** [comment]
You'd need to add `onModifyMovePriority` to the TS definition.

**Comment by Zarel** [comment]
As for `isMetagamiate`, probably just manually `@ts-ignore`.

**Comment by Spandamn** [comment]
for `onModifyMovePriority` is this here?
https://github.com/Zarel/Pokemon-Showdown/blob/538ac5ca5559fbf788becf2472562f29f5022f09/dev-tools/globals.ts#L629

File: chat-plugins/othermetas.js [hunk 4]
```diff
@@ -311,24 +311,22 @@ exports.commands = {
 	},
 	tiershifthelp: [`/ts OR /tiershift <pokemon> - Shows the base stats that a Pokemon would have in Tier Shift.`],
 
-	'!scalemons': true,
-	scale: 'scalemons',
-	scalemons: function (target, room, user) {
```
Comment (KrisXV) [inline chat-plugins/othermetas.js]: Uh no???????????

File: config/formats.js [hunk 5]
```diff
@@ -420,65 +420,62 @@ let Formats = [
...
 			}
-			for (const family in families) {
-				if (families[family].length > 1) problems.push(`${Chat.toListString(families[family])} are in the same evolutionary family.`);
+		},
+		onBasePowerPriority: 8,
+		onBasePower: function (basePower, attacker, defender, move) {
+			if (!move.isMetagamiate) return;
+			return this.chainModify([0x1333, 0x1000]);
+		},
+		validateSet: function (set, teamHas) {
```
Comment (KrisXV) [inline config/formats.js]: Just use onModifyTemplate for this..

File: config/formats.js [hunk 6]
```diff
@@ -420,65 +420,62 @@ let Formats = [
...
-					if (pokemon.prevo) {
-						pokemon = this.getTemplate(pokemon.prevo);
-					}
-				}
-				if (!families[pokemon.species]) families[pokemon.species] = [];
-				families[pokemon.species].push(set.species);
+		mod: 'gen7',
+		ruleset: ['[Gen 7] OU'],
+		banlist: ['Dragonite', 'Kyurem-Black'],
+		onModifyMovePriority: -2,
```
Comment (KrisXV) [inline config/formats.js]: This should be set to -1

File: chat-plugins/othermetas.js (hunk 4 above)
Comment (KrisXV) [reply chat-plugins/othermetas.js]: Add a new command, don’t replace the old one. That’s the whole reason why we havd this chat plugin.

File: config/formats.js [hunk 7]
```diff
@@ -420,65 +420,62 @@ let Formats = [
...
+				let type = types.length < 2 || !pokemon.set.shiny ? types[0] : types[1];
+				move.type = type;
+				move.isMetagamiate = true;
 			}
-			for (const family in families) {
-				if (families[family].length > 1) problems.push(`${Chat.toListString(families[family])} are in the same evolutionary family.`);
+		},
+		onBasePowerPriority: 8,
+		onBasePower: function (basePower, attacker, defender, move) {
+			if (!move.isMetagamiate) return;
```
Comment (KrisXV) [inline config/formats.js]: This is a redundant check, just check if the move.isMetagamiate is set to true

File: config/formats.js (hunk 6 above)
Comment (KrisXV) [reply config/formats.js]: I’m just saying this because you didn’t even remove the -ate checks

Comment (Spandamn) [reply config/formats.js]: woops

File: config/formats.js (hunk 5 above)
Comment (Spandamn) [reply config/formats.js]: will it affect validation though?

File: config/formats.js [hunk 8]
```diff
@@ -441,8 +441,7 @@ let Formats = [
 		onBasePowerPriority: 8,
 		onBasePower: function (basePower, attacker, defender, move) {
 			// @ts-ignore
-			if (!move.isMetagamiate) return;
-			return this.chainModify([0x1333, 0x1000]);
+			if (move.isMetagamiate) this.chainModify([0x1333, 0x1000]);
```
Comment (KrisXV) [inline config/formats.js]: you still need to ``return``

File: config/formats.js (hunk 5 above)
Comment (KrisXV) [reply config/formats.js]: if you use `template = Object.assign({}, template);` with similar event looping code, then no

File: config/formats.js (hunk 1 above)
Comment (urkerab) [reply config/formats.js]: On second thoughts, leave this as `-1` (unless you want to change Electrify and Ion Deluge).

**Comment by Spandamn** [comment]
@KrisXV is doing it because blah blah typescript
watch n learn spandamn

**Pull request closed without merging**